BraveWallet/Extensions/BraveWalletSwiftUIExtensions.swift

```diff
@@ -24,27 +24,36 @@ extension BraveWallet.EthereumChain: Identifiable {
   public var id: String {
     chainId
   }
+  
+  public var nativeToken: BraveWallet.ERCToken {
+    .init(contractAddress: "",
+          name: symbolName,
+          logo: iconUrls.first ?? "",
+          isErc20: false,
+          isErc721: false,
+          symbol: symbol,
+          decimals: decimals,
+          visible: false,
+          tokenId: ""
+    )
+  }
 }
 
 extension BraveWallet.ERCToken: Identifiable {
   public var id: String {
     symbol.lowercased()
   }
-  /// Whether or not this ERCToken is actually ETH
-  public var isETH: Bool {
-    contractAddress.isEmpty && symbol.lowercased() == "eth"
-  }
   
-  public func contractAddress(in chainId: String) -> String {
-    if chainId == BraveWallet.RopstenChainId {
+  public func contractAddress(in network: BraveWallet.EthereumChain) -> String {
+    if network.chainId == BraveWallet.RopstenChainId {
       switch symbol.uppercased() {
       case "ETH": return BraveWallet.ethSwapAddress
       case "DAI" : return BraveWallet.daiSwapAddress
       case "USDC": return BraveWallet.usdcSwapAddress
       default: return contractAddress
       }
     } else {
-      return isETH ? BraveWallet.ethSwapAddress : contractAddress
+      return symbol == network.symbol ? BraveWallet.ethSwapAddress : contractAddress
```

Comment:
Do we need this to be a more concrete check for `eth` or do non-ETH chains use the same `0xeee…` swap address for their native coins?

Reply:
> Do we need this to be a more concrete check for `eth` or do non-ETH chains use the same `0xeee…` swap address for their native coins?

no buy or swap for custom networks :p

Reply:
Oh that's right hah. Can you drop a comment about that here please :)

Reply:
Done!